### PR TITLE
Update Nodejs engines field to >=12

### PR DESCRIPTION
**Pull request by carlo-808**
## Proposed Release Notes
* Added check for minimum `node` version >= 12.
* Set package.json engines `node` field >= 12 and `npm` field to >=6.
## Links
Closes: #797

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -48,16 +48,16 @@ function initialize() {
       preAgentTime
     )
 
-    // TODO: Update this check when Node v10 is deprecated.
-    if (psemver.satisfies('<10.0.0')) {
+    // TODO: Update this check when Node v12 is deprecated.
+    if (psemver.satisfies('<12.0.0')) {
```
Comment (bizob2828) [inline index.js:52]: i feel like this series of checks could be simplified:
 * if !semver sastisfied: requires version based on engines
 * if >= 17.0.0: untested version

thoughts?

File: package.json [hunk 2]
```diff
@@ -111,7 +111,7 @@
   ],
   "homepage": "http://github.com/newrelic/node-newrelic",
   "engines": {
-    "node": ">=10.0.0",
+    "node": ">=12.0.0",
     "npm": ">=3.0.0"
```
Comment (bizob2828) [inline package.json:115]: should we bump this to align with npm 6 as that's what is shipped with 12?

File: index.js (hunk 1 above)
Comment (carlo-808) [reply index.js:52]: I see what you mean and I agree. I'll rework this.

File: package.json (hunk 2 above)
Comment (carlo-808) [reply package.json:115]: Seems fine to me since all these changes require a major version bump.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**